src/lingrex/trimming.py

```diff
@@ -1,201 +1,198 @@
 """
-Trimming functionalities in lingrex.
+Functionality to trim alignments by removing sites.
 """
 import random
+import typing
+import functools
+import itertools
 import collections
-from lingpy.sequence.sound_classes import token2class
-
 
-def revert(alms):
-    return [[row[i] for row in alms] for i in range(len(alms[0]))]        
+from lingpy.sequence.sound_classes import token2class
 
+from lingrex.util import subsequence_of
 
-def get_skeleton(alms, gap="-"):
-    return [token2class(([c for c in col if c != gap] or ["+"])[0], "cv") for col in revert(alms)]
+__all__ = ['GAP', 'Site', 'Sites', 'prep_alignments']
+GAP = '-'
 
 
-def apply_trim(alms, idxs):
+class Site(list):
     """
-    Basic trimming function, based on a selection of indices that are trimmed.
+    A site in an alignment is a "column", i.e. a list of the n-th sound in the aligned words.
     """
-    return [[row[i] for i in range(len(row)) if i not in idxs] for row in alms]
+    def gap_ratio(self, gap: str = GAP) -> float:
+        return self.count(gap) / len(self)
 
+    def first_sound(self, gap=GAP):
+        for s in itertools.dropwhile(lambda c: c == gap, self):
+            return s
 
-def subsequence_of(source, target):
-    """
-    Check if source is a subsequence of target.
-    """
-    q_1, q_2 = list(target), list(source)
-    while q_1:
-        s = q_1.pop(0)
-        if q_2 and q_2[0] == s:
-            q_2.pop(0)
-        elif q_2:
-            pass
-        else:
-            break
-    if not q_2:
-        return True
-    return False
+    def soundclass(self, gap: str = GAP) -> str:
+        return token2class(self.first_sound(gap=gap) or "+", "cv")
 
 
-def consecutive_gaps(seq, gap="-"):
+class Sites(list):
     """
-    Return consecutive gaps in line.
+    A Sites object represents an alignment in the orthogonal view, i.e. listing columns rather
+    than rows.
+
+    .. code-block:: python
+
+        >>> s = Sites(['s-terb-', 'mete---', '-ate-bu', '--te-b-'])
+        >>> print(s)
+        s	-	t	e	r	b	-
+        m	e	t	e	-	-	-
+        -	a	t	e	-	b	u
+        -	-	t	e	-	b	-
+        >>> print(s.trimmed(strategy='gap-oriented'))
+        t	e	b
+        t	e	-
+        t	e	b
+        t	e	b
+        >>> print(s.trimmed(strategy='core-oriented'))
+        t	e	r	b
+        t	e	-	-
+        t	e	-	b
+        t	e	-	b
```

Comment:
I think it's a nice touch if the example from the paper can be re-created transparently here.